From b117b628a06ab740e3e813943d8ef5347cb9e1db Mon Sep 17 00:00:00 2001
From: Chia-Wei Wang <chiawei_wang@aspeedtech.com>
Date: Tue, 7 Dec 2021 10:45:51 +0800
Subject: [PATCH 03/11] ipmi: kcs: aspeed-g6: Use v1 binding

Make KCS driver adopt v1 binding as the v2 one
canno be applied on KCS over PCIe.

Signed-off-by: Chia-Wei Wang <chiawei_wang@aspeedtech.com>
Change-Id: I37ff94c1611a520e84467b0d9cf7615432b5cdb1
---
 arch/arm/boot/dts/aspeed-g6.dtsi   | 16 ++++++++--------
 drivers/char/ipmi/kcs_bmc_aspeed.c |  3 ++-
 2 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed-g6.dtsi
index 3765d2567f1a..00ac85dbd2b0 100644
--- a/arch/arm/boot/dts/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed-g6.dtsi
@@ -562,8 +562,8 @@
 				ranges = <0x0 0x1e789000 0x1000>;
 
 				kcs1: kcs@24 {
-					compatible = "aspeed,ast2500-kcs-bmc-v2";
-					reg = <0x24 0x1>, <0x30 0x1>, <0x3c 0x1>;
+                    compatible = "aspeed,ast2600-kcs-bmc";
+                    reg = <0x0 0x130>;
 					interrupts = <GIC_SPI 138 IRQ_TYPE_LEVEL_HIGH>;
 					clocks = <&syscon ASPEED_CLK_GATE_LCLK>;
 					kcs_chan = <1>;
@@ -571,24 +571,24 @@
 				};
 
 				kcs2: kcs@28 {
-					compatible = "aspeed,ast2500-kcs-bmc-v2";
-					reg = <0x28 0x1>, <0x34 0x1>, <0x40 0x1>;
+                    compatible = "aspeed,ast2600-kcs-bmc";
+                    reg = <0x0 0x130>;
 					interrupts = <GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>;
 					clocks = <&syscon ASPEED_CLK_GATE_LCLK>;
 					status = "disabled";
 				};
 
 				kcs3: kcs@2c {
-					compatible = "aspeed,ast2500-kcs-bmc-v2";
-					reg = <0x2c 0x1>, <0x38 0x1>, <0x44 0x1>;
+                    compatible = "aspeed,ast2600-kcs-bmc";
+                    reg = <0x0 0x130>;
 					interrupts = <GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>;
 					clocks = <&syscon ASPEED_CLK_GATE_LCLK>;
 					status = "disabled";
 				};
 
 				kcs4: kcs@114 {
-					compatible = "aspeed,ast2500-kcs-bmc-v2";
-					reg = <0x114 0x1>, <0x118 0x1>, <0x11c 0x1>;
+                    compatible = "aspeed,ast2600-kcs-bmc";
+                    reg = <0x0 0x130>;
 					interrupts = <GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>;
 					clocks = <&syscon ASPEED_CLK_GATE_LCLK>;
 					status = "disabled";
diff --git a/drivers/char/ipmi/kcs_bmc_aspeed.c b/drivers/char/ipmi/kcs_bmc_aspeed.c
index b555286016b1..a0c802dc7ff2 100644
--- a/drivers/char/ipmi/kcs_bmc_aspeed.c
+++ b/drivers/char/ipmi/kcs_bmc_aspeed.c
@@ -691,9 +691,10 @@ static const struct aspeed_kcs_of_ops of_v2_ops = {
 static const struct of_device_id ast_kcs_bmc_match[] = {
 	{ .compatible = "aspeed,ast2400-kcs-bmc", .data = &of_v1_ops },
 	{ .compatible = "aspeed,ast2500-kcs-bmc", .data = &of_v1_ops },
+    { .compatible = "aspeed,ast2600-kcs-bmc", .data = &of_v1_ops },
 	{ .compatible = "aspeed,ast2400-kcs-bmc-v2", .data = &of_v2_ops },
 	{ .compatible = "aspeed,ast2500-kcs-bmc-v2", .data = &of_v2_ops },
-	{ .compatible = "aspeed,ast2600-kcs-bmc", .data = &of_v2_ops },
+    { .compatible = "aspeed,ast2600-kcs-bmc-v2", .data = &of_v2_ops },
 	{ }
 };
 MODULE_DEVICE_TABLE(of, ast_kcs_bmc_match);
-- 
2.17.1

